runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ParDoTranslationTest.java

```diff
@@ -143,22 +143,33 @@ public void toAndFromTransformProto() throws Exception {
       inputs.putAll(parDo.getAdditionalInputs());
       PCollectionTuple output = mainInput.apply(parDo);
 
-      SdkComponents components = SdkComponents.create();
-      String transformId =
-          components.registerPTransform(
+      SdkComponents sdkComponents = SdkComponents.create();
+
+      // Encode
+      for (PValue input : inputs.values()) {
```

Comment:
I simultaneously want a test that demonstrates our ability to register only a `PTransform` and get the closure of things referred to by that `PTransform` - that's likely to be useful in some form of mix-'n-match; however, decoding these things may not be possible.